pkgs/by-name/vu/vulkan-memory-allocator/package.nix

```diff
@@ -0,0 +1,31 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+}:
+
+stdenv.mkDerivation (finalAttrs: {
```

Comment:
By looking at the output I have noticed this package produces only text files: a C header and a Cmake file.

Does it require C compiler?

Reply:
It does. Using stdenvNoCC gives an error:

```
-- The CXX compiler identification is unknown
CMake Error at CMakeLists.txt:3 (project):
  No CMAKE_CXX_COMPILER could be found.

  Tell CMake where to find the compiler by setting either the environment
  variable "CXX" or the CMake cache entry CMAKE_CXX_COMPILER to the full path
  to the compiler, or to the compiler name if it is in the PATH.
```

Reply:
That's because the CMake script *says* that it needs one:
https://github.com/GPUOpen-LibrariesAndSDKs/VulkanMemoryAllocator/blob/257138b8f5686ae84491a3df9f90a77d5660c3bd/CMakeLists.txt#L3

If you patch this to say `LANGUAGES NONE` instead, then building with `stdenvNoCC` works fine:
```
Running phase: buildPhase
build flags: -j40 SHELL=/nix/store/m101dg80ngyjdb02g6jwy80sr7kzj26g-bash-5.2p26/bin/bash
Running phase: installPhase
install flags: -j40 SHELL=/nix/store/m101dg80ngyjdb02g6jwy80sr7kzj26g-bash-5.2p26/bin/bash install
Install the project...
-- Install configuration: "Release"
-- Installing: /nix/store/gmj0mdqfqf499f9mpg2fbmicqz4zfcdm-vulkan-memory-allocator-3.1.0/include
-- Installing: /nix/store/gmj0mdqfqf499f9mpg2fbmicqz4zfcdm-vulkan-memory-allocator-3.1.0/include/vk_mem_alloc.h
-- Installing: /nix/store/gmj0mdqfqf499f9mpg2fbmicqz4zfcdm-vulkan-memory-allocator-3.1.0/share/cmake/VulkanMemoryAllocator/VulkanMemoryAllocatorConfig.cmake
Running phase: fixupPhase
```

Reply:
Patched. Unfortunately it looks like we can't submit this upstream because [the samples](https://github.com/GPUOpen-LibrariesAndSDKs/VulkanMemoryAllocator/blob/257138b8f5686ae84491a3df9f90a77d5660c3bd/src/CMakeLists.txt) require a compiler